### PR TITLE
feat: allow user to use any HF model

### PR DESCRIPTION
**Pull request by antonioloison**
POC for using any pre-trained model:

Currently, with few changes, it only works with HuggingFace's Hub Fill Mask models in Pytorch.

What was tested and doesn't work:
- `kiliautoml train --api-key $API_KEY --project-id $PROJECT_ID --model-name ../pretrained-model` --> can't use custom pretrained models
- `kiliautoml train --api-key $API_KEY --project-id $PROJECT_ID --model-name jplu/tf-xlm-r-ner-40-lang --ml-backend tensorflow` --> the model doesn't have the right output
- `kiliautoml train --api-key $API_KEY --project-id $PROJECT_ID --model-name Jean-Baptiste/roberta-large-ner-english` --> for sequence classification models, the output doesn't have the right format

### CODE REVIEW

File: kiliautoml/models/_base_model.py [hunk 1]
```diff
@@ -119,7 +135,7 @@ def is_compatible(self, cdt_requested: ModelConditionsRequested) -> bool:
         if strict_conditions and tools_ok:
             # We then check the loose conditions
             self._check_compatible(cdt_requested.ml_backend, self.possible_ml_backend, "ml_backend")
-            self._check_compatible(cdt_requested.model_name, self.advised_model_names, "model_name")
+            self._check_compatible_model(cdt_requested.model_name)
```
Comment (PierreLeveau) [inline kiliautoml/models/_base_model.py:138]: I propose the following UX: we should still advise models. In this case:
* we outputting a warning saying that the model is not in the advised list.
* provide a CLI command that lists the advised models for a given Kili project.
WDYT?

Comment (antonioloison) [reply kiliautoml/models/_base_model.py:138]: Yes, that could be really nice. I will work on that.

File: kiliautoml/models/_base_model.py [hunk 2]
```diff
@@ -105,6 +108,24 @@ def _check_compatible(request: Optional[T], list_ok: List[T], param_name: str) -
                 f"You requested {param_name} {request} but only {list_ok} is available."
             )
 
+    def _check_compatible_model(self, model_name: Optional[ModelNameT]) -> None:
```
Comment (PierreLeveau) [inline kiliautoml/models/_base_model.py:111]: this should be an abstract method of the BaseModel, with this implementation in the HuggingFace models. Ping me if you need help on this.

File: kiliautoml/models/_base_model.py [hunk 3]
```diff
@@ -145,12 +166,14 @@ def __init__(self, base_init_args: BaseInitArgs) -> None:
             self.model_conditions.model_repository,
         )
 
-        self.model_name = set_default(
-            base_init_args["model_name"],
-            self.model_conditions.advised_model_names[0],
-            "model_name",
-            self.model_conditions.advised_model_names,
-        )
+        if "huggingface" not in self.model_repository_dir:
```
Comment (PierreLeveau) [inline kiliautoml/models/_base_model.py:169]: same as above

Comment (PierreLeveau) [reply kiliautoml/models/_base_model.py:169]: (base_model method should have a method `fill_model_name` doing:
```
self.model_name = set_default(
                base_init_args["model_name"],
                self.model_conditions.advised_model_names[0],
                "model_name",
                self.model_conditions.advised_model_names,
            )
            ```
but the HF models should override it with a method that does nothing. The `BaseModel` should not mention HF at all.

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: commands/label_errors.py [hunk 4]
```diff
@@ -290,9 +290,8 @@ def main(
             id_array=[a.id for a in assets],
             errors_by_asset=[[] for _ in assets],
         )
-        model = KiliAutoModel(
-            base_init_args=base_init_args, condition_requested=condition_requested
-        )
+        appropriate_model = get_appropriate_model(condition_requested)
+        model = KiliAutoModel(base_init_args=base_init_args, model_type=appropriate_model)
```
Comment (PierreLeveau) [inline commands/label_errors.py:294]: `model_class`

File: commands/label_errors.py [hunk 5]
```diff
@@ -290,9 +290,8 @@ def main(
             id_array=[a.id for a in assets],
             errors_by_asset=[[] for _ in assets],
         )
-        model = KiliAutoModel(
-            base_init_args=base_init_args, condition_requested=condition_requested
-        )
+        appropriate_model = get_appropriate_model(condition_requested)
```
Comment (PierreLeveau) [inline commands/label_errors.py:293]: get_matching_model_class

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
* get rid of KiliAutoModel -> replace by a function
* add a new command: advise

File: commands/advise.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    api_key: str,
+    ml_backend: MLBackendT,
+    model_name: ModelNameT,
+    model_repository: ModelRepositoryT,
+    project_id: ProjectIdT,
+    target_job: List[JobNameT],
+    ignore_job: List[JobNameT],
+    verbose: VerboseLevelT,
+):
+    """Train a model and then save the model in the cache.
```
Comment (PierreLeveau) [inline commands/advise.py:47]: change docstring

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

**Review by PierreLeveau — APPROVED** [review]
